File: planet-render/src/main/java/au/com/codeka/warworlds/planetrender/TemplatedVoronoi.java
package au.com.codeka.warworlds.planetrender;

import au.com.codeka.warworlds.common.Voronoi;
import java.util.Random;

/** A {@link Voronoi} that takes it's parameters from a {@link Template}. */
public class TemplatedVoronoi extends Voronoi {
  public TemplatedVoronoi(Template.VoronoiTemplate tmpl, Random rand) {
    super(new TemplatedPointCloud(tmpl.getParameter(Template.PointCloudTemplate.class), rand));
  }
}

File: planet-render/src/main/kotlin/au/com/codeka/warworlds/planetrender/TemplatedVoronoi.kt
package au.com.codeka.warworlds.planetrender

import au.com.codeka.warworlds.common.Voronoi
import au.com.codeka.warworlds.planetrender.Template.PointCloudTemplate
import au.com.codeka.warworlds.planetrender.Template.VoronoiTemplate
import java.util.*

/** A [Voronoi] that takes it's parameters from a [Template].  */
class TemplatedVoronoi(tmpl: VoronoiTemplate, rand: Random)
  : Voronoi(TemplatedPointCloud(tmpl.getParameter(PointCloudTemplate::class.java)!!, rand))